src/bootstrap/doc.rs

```diff
@@ -669,16 +669,19 @@ macro_rules! tool_doc {
                     cargo.arg("-p").arg($krate);
                 )+
 
-                $(if !$bin {
+                if !tool_doc!(@expr_or_false, $($bin)?) {
                     cargo.rustdocflag("--document-private-items");
-                })?
+                }
                 cargo.rustdocflag("--enable-index-page");
                 cargo.rustdocflag("--show-type-layout");
                 cargo.rustdocflag("-Zunstable-options");
                 builder.run(&mut cargo.into());
             }
         }
-    }
+    };
+
+    (@expr_or_false,) => { false };
+    (@expr_or_false, $e:expr) => { $e };
```

Comment:
I'd prefer to not get fancy with macros like this -- can we just make the binary "argument" required and update the appropriate places?

Reply:
Understandable! It's fixed now.